.github/workflows/rust.yml

```diff
@@ -6,6 +6,20 @@ jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v1
+    - uses: actions/checkout@v2
+
+    - name: Cargo Home Cache
+      uses: actions/cache@v2
+      env:
+        cache-name: cargo-home
+      with:
+        path:|
+          .cargo
```

Comment:
Do we want this whole directory?  Here's the recommended Rust setup: https://github.com/actions/cache/blob/main/examples.md#rust---cargo

Maybe https://github.com/marketplace/actions/rust-cache is worth investigating?  It offers an arbitrary `key` that it seems like you could use as a buster.